### ISSUE TITLE
blocked-uri in CSP report should be the request URI, not the URI after the redirect

### ISSUE THREAD

**Issue by ckerschb** [web-platform-tests/wpt#27384]
https://github.com/web-platform-tests/wpt/blob/1a7d6fcf01f28389b557b20951808a8adfef7e2d/content-security-policy/frame-src/frame-src-redirect.html#L24-L26

According to the spec the `blocked-uri` should be the original URI, not the URI that got blocked after the redirect (https://www.w3.org/TR/CSP/#security-violation-reports).

@mikewest

**Comment by annevk** [comment]
Ideally we'd also always compare the full values rather than just their origin.

cc @antosart

**Comment by antosart** [comment]
Yes, we should fix the implementations and the tests here. I opened a bug for chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=1172898

**Comment by antosart** [comment]
@annevk wouldn't we leak cross-origin information if we use the full blocked-uri in the report? The report is sent to the parent frame, while the child frame might have been navigated to a URL which the parent does not know and must not know.

**Comment by annevk** [comment]
I'm not sure we should report anything at all. It seems rather problematic that the parent would get access to such information. I guess even if we reported nothing the parent could go on a fishing expedition, but at least that would take a long time and likely not be effective.

(I think I would have preferred if only the initial navigation could be controlled and any further navigations would not be subject to CSP at all.)
